src/server/svr_func.c

```diff
@@ -4717,6 +4720,30 @@ is_runnable(job *ptr, struct prov_vnode_info *pvnfo)
 				break;
 			}
 		}
+		if (((mom_svrinfo_t *)(np->nd_moms[0]->mi_data))->msr_is_hook_checksums_done)
+			 /* count nodes who have processed IS_HOOK_CHECKSUMS message */
+			syncnodes++;
+	}
+	if (!eflag 
+		&& (syncnodes != num_of_prov_vnodes
+		|| do_sync_mom_hookfiles
+		|| sync_mom_hookfiles_proc_running)) {
+
+		/* check if IS_HOOK_CHECKSUMS is processed if not then
+		 * come back after 5 sec and check again, this affirms
+		 * the hook files are in sync before starting the job 
+		 **/
+
+		DBPRT(("%s: node:%s not yet processed IS_HOOK_CHECKSUMS\n",
+			__func__, pvnfo->pvnfo_vnode))
+
+		/* set a work task to come back here after 5 sec */ 
+		if (!set_task(WORK_Timed, time_now + 5, 
+			is_sync_hooks_done, pvnfo)){
+			log_err(errno, "is_sync_hooks_done",
+				"Unable to set task for is_sync_hooks_done");
+		}
+		eflag = -5;
```

Comment:
The purpose of new value -5 is, this will be returned when all nodes are up and hooks are not in sync also indicates that timed task has been created so that we need not remove the provisioning record, as in timed task we are calling directly check_and_run_job which is accessing this record to get the job info.
In the case of -1, it is returned when node state is in INUSE_PROV means it will come back again when the node is rebooted and we are removing provisioning record so it will retrieve again from is_vnode_prov_done and that will call check_and_run_job.
We can use -1 but need to retrieve provisioning record from provisioning table before calling check_and_run_job.